langchain_benchmarks/rate_limiting.py

```diff
@@ -0,0 +1,107 @@
+"""Implementation of a rate limiter based on a token bucket."""
+import threading
+import time
+from typing import Any, Optional
+
+from langchain.schema.runnable import Runnable, RunnableLambda
+from langchain.schema.runnable.utils import Input, Output
+
+
+class RateLimiter:
+    def __init__(
+        self,
+        *,
+        requests_per_second: float = 1,
+        tokens_per_request: int = 1,
```

Comment:
Given our domain, could we clarify that tokens are NOT LLM tokens? I could imagine someone thinking this being the number of ~words we can send to the LLM per request. Maybe link to the token bucket algorithm for complete novices as well.

This is especially the case given that some providers rate limit based on e.g., prompt tokens or total tokens per minute/hour

Reply:
Similarly, should call `requests_per_second` something like `tokens_per_second` to make it more clear?